### PR TITLE
Stricter pull_request_target in labeler.yml

### PR DESCRIPTION
**Pull request by kit1980**
Copied from https://github.com/pytorch/pytorch/blob/main/.github/workflows/check-labels.yml

### CODE REVIEW

**Review by izaitsevfb — APPROVED** [review]

File: .github/workflows/labeler.yml [hunk 1]
```diff
@@ -1,7 +1,21 @@
...
 
 on:
-- pull_request_target
+  # We need pull_request_target to be able to add labels to PRs from forks.
+  # Only allow pull_request_target when targeting main, not some historical branch.
+  #
+  # Make sure to don't introduce explicit checking out and installing/running
+  # untrusted user code into this workflow!
+  pull_request_target:
+    types: [opened, synchronize, reopened, labeled, unlabeled]
```
Comment (izaitsevfb) [inline .github/workflows/labeler.yml:10]: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target

> By default, a workflow only runs when a pull_request_target event's activity type is opened, synchronize, or reopened

Are you sure that we need to add `labeled`, `unlabeled` here?

Comment (kit1980) [reply .github/workflows/labeler.yml:10]: Let's remove.

**Review by malfet — APPROVED** [review]

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**